Source/Diagnostics/WarpXOpenPMD.cpp

```diff
@@ -0,0 +1,479 @@
+#include "WarpXOpenPMD.H"
+#include "FieldIO.H"  // for getReversedVec
+
+WarpXOpenPMDPlot::WarpXOpenPMDPlot(bool oneFilePerTS,
+                       const std::string& openPMDFileType)
```

Comment:
In C++11, the interface can just be `const std::string openPMDFileType` to make use of copy elision guarantees. Consequently, below ...